src/Rule.php

```diff
@@ -2135,10 +2135,13 @@ public function getNextRanking()
     {
         global $DB;
 
+        // Try to fix unregulated addition via legacy API at /Rule endpoint instead of /Rule{Type}
+        // Use the sub_type field if it exists and is not the same as the current class. Otherwise, use the current class at runtime.
+        $sub_type = (isset($this->fields['sub_type']) && $this->fields['sub_type'] !== self::class) ? $this->fields['sub_type'] : static::class;
```

Comment:
Maybe a new `?string $sub_type = null` param should be added to current method, to prevent this kind of "hack". In `prepareInputForAdd()`, it would require to move the call to `getNextRanking()` after the whole code related to `$sub_type` computation.